src/Cmfd.cpp

```diff
@@ -1872,71 +1476,71 @@ void Cmfd::generateKNearestStencils(){
  * @param fsr The fsr being updated.
  * @return the ratio used to update the FSR flux.
  */
-FP_PRECISION Cmfd::getUpdateRatio(int cell, int group, int fsr){
+FP_PRECISION Cmfd::getUpdateRatio(int cell, int group, int fsr) {
 
   FP_PRECISION ratio = 0.0;
   FP_PRECISION total_distance;
   std::vector< std::pair<int, FP_PRECISION> >::iterator iter;
 
-  if (_centroid_update_on){
+  if (_centroid_update_on) {
 
     total_distance = _centroid_total_distances[fsr];
```

Comment:
Instead of storing the distances, have you considered storing ``iter->second/total_distance`` to avoid lots of expensive, redundant divisions? Or maybe even better yet for code simplicity, store (1.0 - iter->second/total_distance). This might even cut into the 15% slowdown in CMFD with *k*-nearest neighbor stencils